pandas/core/groupby.py

```diff
@@ -4580,7 +4580,7 @@ def _get_sorted_data(self):
 
         # this is sort of wasteful but...
         sorted_axis = data.axes[self.axis].take(self.sort_idx)
-        sorted_data = data.reindex(sorted_axis, axis=self.axis)
+        sorted_data = data.reindex_axis(sorted_axis, axis=self.axis)
```

Comment:
This won't cause a deprecation warning in our own code?

Reply:
`data` here is a BlockManager (I incorrectly changed this in an earlier commit)